src/plugins/intellisense/index.tsx

```diff
@@ -0,0 +1,512 @@
+          const myindex = orderMap.get(selectedId) ?? 0;
+          this.intellisenseOpts.sort((a, b) => {
+            return (
+              Math.abs((orderMap.get(a.exprId) ?? 0) - myindex) -
+              Math.abs((orderMap.get(b.exprId) ?? 0) - myindex)
+            );
+          });
+        }
+
+        if (this.intellisenseIndex === -1) this.intellisenseIndex = 0;
+      } else {
+        this.intellisenseOpts = [];
+        this.intellisenseIndex = -1;
+      }
+
+      // if there isn't, just get rid of the intellisense window
+    } else {
+      this.intellisenseOpts = [];
+      this.latestIdent = undefined;
+      this.latestMQ = undefined;
+      this.intellisenseIndex = -1;
+    }
+
+    // update intellisense window
+    this.view?.update();
+  }
+
+  // leave an intellisense menu and return to whatever expression
+  // you were previously in
+  leaveIntellisenseMenu() {
+    if (this.intellisenseReturnMQ) {
+      this.intellisenseReturnMQ?.focus();
+      this.latestMQ = this.intellisenseReturnMQ;
+    }
+
+    if (this.prevCursorPos && this.latestMQ) {
+      const mqRootBlock =
+        this.latestMQ.__controller.container.querySelector(
+          ".dcg-mq-root-block"
+        );
+
+      if (!mqRootBlock) return;
+
+      const mqRootBlockRect = mqRootBlock.getBoundingClientRect();
+
+      // simulate a click to get cursor in the right spot
+      const eventHandlerSettings = {
+        bubbles: true,
+        clientX:
+          this.prevCursorPos.x - mqRootBlock.scrollLeft + mqRootBlockRect.x,
+        clientY:
+          this.prevCursorPos.y - mqRootBlock.scrollTop + mqRootBlockRect.y,
+      };
+      this.latestMQ.__controller.container.dispatchEvent(
+        new MouseEvent("mousedown", eventHandlerSettings)
+      );
+      this.latestMQ.__controller.container.dispatchEvent(
+        new MouseEvent("mouseup", eventHandlerSettings)
+      );
+    }
+  }
+
+  // keep track of where the cursor is so we can return to it
+  // once we refocus the mathquill input
+  saveCursorState() {
+    const focusedmq = MathQuillView.getFocusedMathquill();
+    if (focusedmq) this.latestMQ = focusedmq;
+    if (
+      this.latestMQ &&
+      document.body.contains(this.latestMQ.__controller.cursor.cursorElement)
+    ) {
+      // get cursor pos relative to the top left of the mathquill's root element
+      const mqRootBlock =
+        this.latestMQ.__controller.container.querySelector(
+          ".dcg-mq-root-block"
+        );
+
+      if (!mqRootBlock) return;
+      const mqRootBlockRect = mqRootBlock.getBoundingClientRect();
+      const rect =
+        this.latestMQ.__controller.cursor.cursorElement.getBoundingClientRect();
+      this.prevCursorPos = {
+        x: rect.x + mqRootBlock.scrollLeft - mqRootBlockRect.x,
+        y: rect.y + mqRootBlock.scrollTop - mqRootBlockRect.y,
+      };
+    }
+  }
+
+  focusOutHandler = (e: FocusEvent) => {
+    if (
+      e.target !== intellisenseMountPoint &&
```

Comment:
I'm kinda surprised this works. Normally you have to check if the mount point is an *ancestor* of the event target.

Reply:
If I remember correctly this code might not actually do anything lol--- I removed the ability for `intellisenseMountPoint` to be focused a while ago. I should refactor this.

Reply:
Fixed